include/list.h

```diff
@@ -326,7 +326,7 @@ typedef struct xLIST
         }                                                                        \
                                                                                  \
         ( pxItemToRemove )->pxContainer = NULL;                                  \
-        ( pxList->uxNumberOfItems )--;                                           \
+        ( ( pxList )->uxNumberOfItems ) -= ( UBaseType_t ) 1U;                   \
```

Comment:
One simplification:
~~~diff
-    ( ( pxList )->uxNumberOfItems ) -= ( UBaseType_t ) 1U; 
+    ( ( pxList )->uxNumberOfItems ) -= 1U; 
~~~
I think that the cast is not necessary. Have a try.

Also try:
~~~diff
-    xNumOfOverflows += ( BaseType_t ) 1;   
+    xNumOfOverflows += 1;   
~~~

Reply:
The problem is that BaseType_t can be a different size from "unsigned", which means this will work fine with some types as BaseType_t but not others. These cases are tricky because compiling only with one type for BaseType_t will not expose all the problems !